android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -0,0 +1,1053 @@
+            return this;
+        }
+        public Builder setView(View view) {
+            this.view = view;
+            return this;
+        }
+
+        public MatchingEngineHelper build() {
+            return new MatchingEngineHelper(this);
+        }
+    }
+
+    public MatchingEngineHelper(final Builder builder) {
+        Log.i(TAG, "MatchingEngineHelper()");
+        mActivity = builder.activity;
+        mView = builder.view;
+        meHelperInterface = builder.meHelperInterface;
+        me = new MatchingEngine(mActivity);
+
+        mFusedLocationClient = LocationServices.getFusedLocationProviderClient(mActivity);
+
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(builder.activity);
+        Resources resources = mActivity.getResources();
+
+        // Reuse the onSharedPreferenceChanged code to initialize anything dependent on these prefs:
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_matching_engine_location_verification));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_app_name));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_app_version));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_org_name));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_default_dme_hostname));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_default_operator_name));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_find_cloudlet_mode));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_app_instances_limit));
+        onSharedPreferenceChanged(prefs, resources.getString(R.string.pref_default_app_definition));
+
+        Log.i(TAG, "mDmeHostname="+ mDmeHostname +" networkSwitchingAllowed="+mNetworkSwitchingAllowed+" mCarrierName="+mCarrierName);
+
+        // Watch for any updated preferences:
+        prefs.registerOnSharedPreferenceChangeListener(this);
+
+        mEdgeEventsSubscriber = new EdgeEventsSubscriber();
+        me.getEdgeEventsBus().register(mEdgeEventsSubscriber);
+
+        // Set a default Edge Events config.
+        // There is also a parameterized version to further customize.
+        EdgeEventsConfig backgroundEdgeEventsConfig = me.createDefaultEdgeEventsConfig();
+        backgroundEdgeEventsConfig.latencyTestType = NetTest.TestType.CONNECT;
+        // This is the internal port, that has not been remapped to a public port for a particular appInst.
+        backgroundEdgeEventsConfig.latencyInternalPort = DEFAULT_LATENCY_PORT;
+        backgroundEdgeEventsConfig.latencyUpdateConfig.maxNumberOfUpdates = 10; // Or Long.MAX_VALUE if you want. Default is 0.
+        backgroundEdgeEventsConfig.latencyUpdateConfig.updateIntervalSeconds = 7; // The default is 30.
+//        backgroundEdgeEventsConfig.latencyThresholdTrigger = 300;

```

Comment:
Why this is commented out ?

Reply:
It was something Leon suggested I add, during debugging. Didn't make a difference for the case I was testing. Will remove or uncomment.